### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by MaxUlysse**
Replace Docker by BioContainers, that should cover both Docker and Singularity.
repalce previous PR #850 that was made to `dev`

## PR checklist

 - [ ] This comment contains a description of changes (with reason)
 - [ ] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
 <!-- TODO nf-core: Write a 1-2 sentence summary of what data the pipeline is for and what it does -->
 **{{ cookiecutter.name }}** is a bioinformatics best-practise analysis pipeline. {{ cookiecutter.description }}
 
-The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It comes with docker containers making installation trivial and results highly reproducible.
+The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses containers from [BioContainers](https://biocontainers.pro/) making installation trivial and results highly reproducible.
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:22]: Do we want to suggest this given that some pipelines may require containers not on Biocontainers? My thoughts are to leave it generic for now and mention it in free text somewhere else if it is indeed the case?

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:22]: Agreed 👍🏻

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:22]: @MaxUlysse ok if we close this PR then?

Comment (MaxUlysse) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:22]: No, there is still the issue on Docker containers, we now (meaning here in DSL2) do not provide containers anymore.
(except specific case).
We mainly use available containers.
But I'm ok for a change from Docker -> Docker and/or Singularity

Comment (ewels) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/README.md:22]: I'd say that even if we're not building the containers, the pipeline still comes with containers..?

```suggestion
The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker / Singularity containers making installation trivial and results highly reproducible.
```

**Review by ewels — APPROVED** [review]

**Pull request merged**